### PR TITLE
Add identity Pattern simplifier: Exists.hs

### LINKED ISSUE #2946: Add identity `Pattern` simplifier
We should add an identity `Pattern` simplifier (which doesn't do any actual simplification) attempt to replace the old `Pattern` simplifier with this new one in the code. By running the tests, we can identify any places which don't actually need to call the simplifier, or any places where we only need a small subset of the simplifier's cases implemented.

### PR DESCRIPTION
**Pull request by andreiburdusa**
Part of #2946 

### Scope:
- Replace the current simplifier with an identity simplifier

### Estimate:
- ~~January 14~~
- ~~January 24~~
- February 8
---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Review by ana-pantilie — CHANGES_REQUESTED** [review]
Running the integration tests twice, once normally and once with `--simplifierx` isn't currently enabled, right? If it's not, this PR needs to enable it. We discussed this is necessary once we merge the first experimental feature.

**Comment by andreiburdusa** [comment]
> Running the integration tests twice, once normally and once with `--simplifierx` isn't currently enabled, right? If it's not, this PR needs to enable it. We discussed this is necessary once we merge the first experimental feature.

All the integration tests? I thought we don't want to do that because it would take too much time. I probably misunderstood

**Comment by ana-pantilie** [comment]
> All the integration tests? I thought we don't want to do that because it would take too much time.

It's necessary, otherwise we might break features which we actually use in practice along the way.

**Comment by ana-pantilie** [comment]
Also, this will probably make Jenkins time out. You'll need to increase the time out period on Jenkins for our CI/integration tests.

**Comment by andreiburdusa** [comment]
> Also, this will probably make Jenkins time out. You'll need to increase the time out period on Jenkins for our CI/integration tests.

Shoud I try making nix run the integration tests twice as well. or just Jenkins?

**Comment by ana-pantilie** [comment]
> Shoud I try making nix run the integration tests twice as well, or just Jenkins?

Hmm, let's just run them on Jenkins for now.

File: kore/src/Kore/Simplify/Exists.hs [hunk 1]
```diff
@@ -70,6 +70,9 @@ import Kore.Rewrite.RewritingVariable (
 import qualified Kore.Simplify.AndPredicates as And (
     simplifyEvaluatedMultiPredicate,
  )
+import Kore.Simplify.Pattern (
+    simplifyPatternId,
```
Comment (ana-pantilie) [inline kore/src/Kore/Simplify/Exists.hs:74]: I think it's best if we move this into `MonadSimplify` as well. This way, we have access to it throughout the simplification code and we won't introduce new module dependencies.

**Review by ana-pantilie — APPROVED** [review]

**Review by JKTKops — APPROVED** [review]

**Pull request merged**